src/categorical_algebra/Slices.jl

```diff
@@ -0,0 +1,133 @@
+module Slices
+export Slice, SliceMorphism,SliceLimit,SliceColimit, pushout_complement
+
+using AutoHashEquals
+using ...GAT
+using ..FreeDiagrams, ..Limits, ..CSets
+using ...Theories: Category
+import ...Theories: dom, codom, compose, id
+import ..Limits: limit, colimit, universal
+import ..FinSets: force, pushout_complement
+
+@auto_hash_equals struct Slice{Ob,Hom}
+  slice::Hom
+end
+
+@auto_hash_equals struct SliceMorphism{Ob, Hom}
```

Comment:
Likewise about the type parameter `Ob` here. Furthermore, because the concrete morphisms types may differ, I think the struct definition should be:

```julia
struct SliceMorphism{Hom, Dom<:Slice, Codom<:Slice}
  dom::Dom
  codom::Codom
  f::Hom
end
```

Also, I think we should rename `SliceMorphism` to `SliceHom`, not for any particularly good reason, but just for consistency with other names in Catlab.